crates/matrix-sdk-common/src/linked_chunk/relational.rs

```diff
@@ -292,6 +292,21 @@ impl<Item, Gap> RelationalLinkedChunk<Item, Gap> {
             }
         }
     }
+
+    /// Return an iterator that yields events of a particular room with no
+    /// particular order.
+    pub fn unordered_events<'a>(&'a self, room_id: &'a RoomId) -> impl Iterator<Item = &'a Item> {
+        self.items.iter().filter_map(move |item_row| {
+            if item_row.room_id == room_id {
+                match &item_row.item {
+                    Either::Item(item) => Some(item),
+                    Either::Gap(..) => None,
+                }
```

Comment:
super light suggestion: `as_variant!`? :-)

Reply:
Thoughts about it, but I decided it wasn't worth it :-(.